examples/simon.py

```diff
@@ -0,0 +1,339 @@
+'''
+---------------------------------
+SIMONS'S ALGORITHM - OVERVIEW
+---------------------------------
+Simon's Algorithm solves the problem of finding a particular value of s when some function f:{0, 1}^n --> {0, 1}^n
+is inputted into the program that follows this rule: "f(x) = f(y) if and only if x (+) y is in the set {0^n, s}"
+---------------------------------
+STEPS OF THE ALGORITHM
+---------------------------------
+1. Begin with two n-qubit registers, each in the |0> state
+2. Apply a Hadamard transform to the first n-qubit register, therefore creating an even superposition of states
+3. The oracle (which encodes values of the function) is queried B_f |x>|y> = |x>|y (+) f(x)>, therefore mapping |0^n> --> |f(x)>
+4. Apply a Hadamard transform to the first n-qubit register
+---------------------------------
+MEASUREMENT
+---------------------------------
+It can be found that for an output string y, then y (dot mod 2) s is always equal to 0, as we can calculate y (dot mod 2) s = 1 occuring with probability = 0
+We get a system of eqautions, which we can use to solve for s (provided y_1, ..., y_(n-1) are linearlly independent)! We measure the string y to be the first n-qubit register
+'''
+
+import cirq
+import random
+import numpy as np
+import copy
+import sympy
+import itertools
+
+# Qubit preparation
+
+number_qubits = #Number of qubits
+
+def main(number_qubits):
+
+    circuit_sampling = number_qubits-1
+
+    #Create the qubits which are used within the circuit
```

Comment:
```suggestion
    # Create the qubits which are used within the circuit
```
Always space between the `#` and the comment.